### PR TITLE
Entire Logout Functionality

### PR DESCRIPTION
**Pull request by hannahsalameh**
Invalidates the user's session, deletes the session ID from the user entity, and removes the cookie that's tracking log in

### CODE REVIEW

File: src/main/java/com/google/plantasy/servlets/LogoutServlet.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
+package com.google.sps.servlets;
+
+import java.io.*;
+import javax.servlet.*;
+import javax.servlet.http.*;
```
Comment (devonwasson) [inline src/main/java/com/google/plantasy/servlets/LogoutServlet.java:5]: please don't import *, only import what you need.

File: src/main/java/com/google/plantasy/servlets/LogoutServlet.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
+package com.google.sps.servlets;
```
Comment (devonwasson) [inline src/main/java/com/google/plantasy/servlets/LogoutServlet.java:1]: wrong package name

File: src/main/java/com/google/plantasy/servlets/LogoutServlet.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import com.google.plantasy.utils.CookieUtils;
+
+@WebServlet("/logout")
+public class LogoutServlet extends HttpServlet {
+DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      Cookie[] cookies = request.getCookies();
+      Entity userEntity = UserUtils.getUserFromCookie(cookies, datastore);
```
Comment (devonwasson) [inline src/main/java/com/google/plantasy/servlets/LogoutServlet.java:22]: Should this be an entity or a User class?

File: src/main/java/com/google/plantasy/servlets/LogoutServlet.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+      datastore.put(userEntity);
+
+      HttpSession session = request.getSession(false);
+      if(session != null){
+          session.invalidate();
+      }
+
+      Cookie cookie = CookieUtils.getCookieFromName(cookies,"SessionID");
+      cookie.setMaxAge(0); 
+      response.addCookie(cookie);
```
Comment (devonwasson) [inline src/main/java/com/google/plantasy/servlets/LogoutServlet.java:33]: could you add a line documenting what this does?

File: src/main/java/com/google/plantasy/servlets/LogoutServlet.java (hunk 3 above)
Comment (hannahsalameh) [reply src/main/java/com/google/plantasy/servlets/LogoutServlet.java:23]: the user class doesnt handle session id's rn so it has to b an entity

Comment (devonwasson) [reply src/main/java/com/google/plantasy/servlets/LogoutServlet.java:23]: gotcha, I would maybe add that to the user Entity in this CL so we can make that change. We shouldnt be introducing new references to the Entity class.

Comment (hannahsalameh) [reply src/main/java/com/google/plantasy/servlets/LogoutServlet.java:23]: ok cool ill edit it

**Pull request merged**